packages/@aws-cdk/core/lib/bundling.ts

```diff
@@ -121,11 +122,22 @@ export class BundlingDockerImage {
       throw new Error('Failed to extract image ID from Docker build output');
     }
 
-    return new BundlingDockerImage(match[1]);
+    const hash = FileSystem.fingerprint(path, { extraHash: JSON.stringify(options) });
+    return new BundlingDockerImage(match[1], hash);
   }
 
   /** @param image The Docker image */
-  private constructor(public readonly image: string) {}
+  private constructor(public readonly image: string, private readonly _imageHash?: string) {}
+
+  /**
+   * Provides the name that the bundler should use to refer to this image when
+   * hashing the bundling configuration with a source or custom asset hash.
+   *
+   * @return The overridden image name if set or image hash name in that order
+   */
+  public get imageHash() {
```

Comment:
I think overriding `toJSON` would be a better approach here.

Reply:
Done.